lib/oban/job.ex

```diff
@@ -325,6 +338,28 @@ defmodule Oban.Job do
     end
   end
 
+  defp put_replace(changeset, value) do
+    case value do
+      replace_args when is_boolean(replace_args) ->
+        replace = get_change(changeset, :replace, [])
+
+        updated =
+          if replace_args do
+            [:args | replace]
+          else
+            Enum.reject(replace, &(&1 == :args))
+          end
+
+        put_change(changeset, :replace, updated)
+
+      nil ->
+        changeset
+
+      _ ->
+        add_error(changeset, :replace_args, "invalid value")
+    end
+  end
```

Comment:
I believe we can simplify this with `update_change`:

```suggestion
  defp put_replace(changeset, value) do
    case value do
      true -> update_change(changeset, :replace, &[:args | &1])
      false -> update_change(changeset, :replace, &(&1 -- [:args]))
      nil -> changeset
      _ -> add_error(changeset, :replace_args, "invalid value")
    end
  end
```

Reply:
I don't think this will work in the case where `:replace` isn't set, as `update_change` only gets applied when the key is set, unless I'm missing something?

Reply:
Good point! Will tidy this up a bit